### ISSUE TITLE
Merge async and not-async implementation 

### ISSUE THREAD

**Issue by shps951023** [mini-software/MiniExcel#601]

**Comment by duszekmestre** [comment]
What do you mean by this? Maybe I can help

**Comment by shps951023** [comment]
@duszekmestre e.g. ExcelOpenXmlSheetWriter.Async.cs and ExcelOpenXmlSheetWriter.cs, their logic and code 99% same, but miniexcel needs duplicate code to use `async`, `Task`, `await ` keyword.
I think these code can be merged.

**Comment by duszekmestre** [comment]
Yes - I tried to do this but I have limited time for doing this. Some code can be reused and in latest PR i refactor one of the method to prepare string values to write them to the writer.

**Comment by duszekmestre** [comment]
I started to reuse code in PR #602 

In the next part I'll try to reuse code between different types of objects on saving.

**Comment by duszekmestre** [comment]
> @duszekmestre ❤thanks, e.g. ExcelOpenXmlSheetWriter.Async.cs and ExcelOpenXmlSheetWriter.cs, their logic and code 99% same, but miniexcel needs duplicate code to use `async`, `Task`, `await ` keyword. I think these code can be merged.

I think it is not possible to completely reuse same code. Async/await method should be written separately. The only thing which is possible is creating same codebase to generate XML strings for both implementations to reuse and easier maintenance.

I saw that different types (data reader, data table, enumerable) are managed separately so many errors are done because of lack of implementation to one of them. This the hardest and most important thing to rewrite.

If you have any other idea please share with me.

**Comment by shps951023** [comment]
Yes, my first idea is we can only maintain `async` method like HttpClient
If sync method just call async method GetAwaiter and GetResult

**Comment by fpt-ian** [comment]
![Image](https://github.com/user-attachments/assets/58349e7d-2887-42c0-8fcf-667c429486c2)

You said please issue for me so here it is 😊

I cannot get merge to work at all btw. I only tried the async version because I was running out of ideas to get the normal one to work.

It lets me download the excel, but the excel is corrupt. I tried many many things including removing the other non-merge fields, wrapping all rows instead of just per section that should merge.

**Comment by michelebastione** [comment]
@fpt-ian If you're still experiencing this problem could you please create a new issue and attach the file that you're not able to work with?
